### PR TITLE
Improve requests error wrapping

### PR DESCRIPTION
**Pull request by aaschaer**
Addresses #152 

I gave all the exceptions very simple messages for now. We could expand on the conversion logic to look at the underlying exceptions and give more helpful messages based on those data if wanted.

### CODE REVIEW

File: globus_sdk/exc.py [hunk 1]
```diff
@@ -141,11 +142,17 @@ class InvalidDocumentBodyError(GlobusError):
     """
 
 
-# Thin wrappers around requests exceptions, so the SDK is API independent.
+# Wrappers around requests exceptions, so the SDK is API independent.
```
Comment (sirosen) [inline globus_sdk/exc.py]: Haha! Indeed, this comment change summarizes the whole PR! :rofl:

**Comment by sirosen** [comment]
Somehow, an attribute which I thought was promised to us by Python is missing in Windows versions of Python3? Not all `Exception` classes have a `message` attribute?

This is totally weird to me, and I'm not sure what we need to do to make sure we don't trip over it.

**Comment by aaschaer** [comment]
It seems python3 removed Exception storing an explicit message field and just uses the arguments passed to it to form a string. I'll adjust the code to reflect this.

However, I am once again dubious that Travis is actually running our tests with python3

**Comment by sirosen** [comment]
Oh dear, I believe that you are correct...
Taken from one of the Py3 builds:
```
Creating /home/travis/build/globus/globus-sdk-python/.venv/lib/python2.7/site-packages/globus-sdk.egg-link (link to .)
```

Let's resolve in a follow-up issue.

**Pull request merged**